packages/mui-lab/src/LoadingButton/LoadingButton.js

```diff
@@ -106,7 +106,7 @@ const LoadingButtonLoadingIndicator = styled('div', {
       left: 14,
     }),
   ...(ownerState.loadingPosition === 'start' &&
-    ownerState.variant === 'text' && {
+    (!ownerState.variant || ownerState.variant === 'text') && {
```

Comment:
I'm might be missing something but this change looks KO to me. If the Button variant is outlined, then the style of the LoadingButton should also have the outlined version, not the text one. Going on step deeper, I would argue that the issue we link to is not a regression but the correct behavior.

Reply:
@oliviertassinari I agree, and I was personally the one that did the change on the previous PR. The problem is that it is a breaking change, and we released it in a minor, so people complained. We can revisit this again in v6, but in my opinion, it is better to let this change land now. Does this make sense?

Reply:
Interesting, I see two topics:

1. How should we handle bug fixes that are breaking changes? Personally, I think that we can emphasize on the bug fix part. My hypotheses is that for any bug fix, there are people relying on the wrong behavior, we still need to be able to fix bugs.
2. Let's say we are working on v6 right now (we are not, but to simplify the problem) what would be the "right" behavior?

Reply:
> How should we handle bug fixes that are breaking changes? Personally, I think that we can emphasize on the bug fix part. My hypothesis is that for any bug fix, there are people relying on the wrong behavior, we still need to be able to fix bugs.

Agree, the only problem I see with this scenario is that, no one complained (or saw as a bug the behavior that we had before), so if we are changing it, we are kind of fixing a non-existing bug, or at least a bug people are not aware of. In any way, if we actually do this, we should document better in the changelog the fact that the expected behavior changed for the component, so that people can catch it earlier.

Reply:
> Let's say we are working on v6 right now (we are not, but to simplify the problem) what would be the "right" behavior? (I'm asking because it's not clear to me)

I would add a default value or the `variant` inside the `LoadingButton`.

Reply:
> I would add a default value or the variant inside the LoadingButton.

I think that we all agree on this one. I think that the core of the question is: Should `theme.components.MuiButton.defaultProps.variant` also impact `MuiLoadingButton` or not?

> we are kind of fixing a non-existing bug

I can create an issue to highlight how the behavior on v5.0.0 was wrong if needed 😁

Reply:
> Should theme.components.MuiButton.defaultProps.variant also impact the default variant of MuiLoadingButton or not

To be honest I am not sure. From user's perspective it should either always impact the prop in the `LoadingButton` or not. At this moment, it impacts only the props that do not have a default value in the `LoadingButton`. This is not intuitive. 

On one hand, impacting the default props for the `LoadingButton` is good, as the component is kind of an extension of the Button. But on the other hand, it looks like a hidden side effect. I think we should create an RFC and decide on this generally for all components and implement it consistently for v6.

Reply:
@mnajdova, @oliviertassinari, just to be clear: is the conclusion here to merge this PR and think of a better solution for v6?

Reply:
@michaldudak sorry yes. I think we should proceed with the PR as it reverts a breaking change that we introduced. I am adding this discussion to our v6 project :)

Reply:
> To be honest I am not sure. From user's perspective it should either always impact the prop in the LoadingButton or not. At this moment, it impacts only the props that do not have a default value in the LoadingButton. This is not intuitive.
> On one hand, impacting the default props for the LoadingButton is good, as the component is kind of an extension of the Button. But on the other hand, it looks like a hidden side effect. I think we should create an RFC and decide on this generally for all components and implement it consistently for v6.

@mnajdova Ok, in this case, I propose we avoid surprises: theme.components.MuiXXX.defaultProps <==> one component. So no side effects on other components.

Now, in practice, what does it mean?

- the expected behavior in #29740 is wrong. Per the above
- we fixed a bug with #28645 "Now we need also change default prop value for MuiLoadingButton separately" guarentees that not only the prop is correct **but also the styles**.

Reply:
Sounds fair, I left a comment on the issue, I am closing this PR too. @ZeeshanTamboli thanks for the effort on this one.